rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -81,150 +90,257 @@ typedef struct
+    virtual JointsData& get_joints_data(){  };
+    ImgSize& get_img_size() {return _info.img_size; }
+    std::string& get_image_name() { return _info.img_name; }
+    uint& get_image_id() { return _info.img_id; }
+    void set_img_size(ImgSize img_size) { _info.img_size = std::move(img_size); }
+    void set_img_id(uint img_id) { _info.img_id = img_id; }
+    void set_img_name(std::string img_name) { _info.img_name = img_name; }
+    void set_metadata_info(MetaDataInfo info) { _info = std::move(info); }
+    protected:
+    MetaDataInfo _info;
 };
 
 struct Label : public MetaData
 {
-    Label(int label) { _label_id = label; }
-    Label(){ _label_id = -1; }
+    Label(int label)
+    {
+        _label_ids = {label};
+    }
+    Label()
+    {
+        _label_ids = {-1};
+    }
+    std::vector<int>& get_labels() { return _label_ids; }
+    void set_labels(Labels label_ids)
+    {
+        _label_ids = std::move(label_ids);
+    }
+    protected:
+    Labels _label_ids = {}; // For label use only
 };
 
-struct BoundingBox : public MetaData
+struct BoundingBox : public Label
 {
     BoundingBox()= default;
-    BoundingBox(BoundingBoxCords bb_cords, BoundingBoxLabels bb_label_ids)
+    BoundingBox(BoundingBoxCords bb_cords, Labels bb_label_ids, ImgSize img_size = ImgSize{0,0}, uint img_id = 0)
     {
         _bb_cords =std::move(bb_cords);
-        _bb_label_ids = std::move(bb_label_ids);
+        _label_ids = std::move(bb_label_ids);
+        _info.img_size = std::move(img_size);
+        _info.img_id = img_id;
     }
-    BoundingBox(BoundingBoxCords bb_cords, BoundingBoxLabels bb_label_ids, ImgSize img_size)
-    {
-        _bb_cords =std::move(bb_cords);
-        _bb_label_ids = std::move(bb_label_ids);
-        _img_size = std::move(img_size);
-    }
-    void set_bb_cords(BoundingBoxCords bb_cords) { _bb_cords =std::move(bb_cords); }
-    BoundingBox(BoundingBoxCords_xcycwh bb_cords_xcycwh, BoundingBoxLabels bb_label_ids)
+    BoundingBox(BoundingBoxCords_xcycwh bb_cords_xcycwh, Labels bb_label_ids, ImgSize img_size = ImgSize{0,0}, uint img_id = 0)
     {
         _bb_cords_xcycwh =std::move(bb_cords_xcycwh);
-        _bb_label_ids = std::move(bb_label_ids);
+        _label_ids = std::move(bb_label_ids);
+        _info.img_size = std::move(img_size);
+        _info.img_id = img_id;
     }
-    BoundingBox(BoundingBoxCords_xcycwh bb_cords_xcycwh, BoundingBoxLabels bb_label_ids, ImgSize img_size)
+    BoundingBoxCords& get_bb_cords() { return _bb_cords; }
+    BoundingBoxCords_xcycwh& get_bb_cords_xcycwh() { return _bb_cords_xcycwh; }
+    void set_bb_cords_xcycwh(BoundingBoxCords_xcycwh bb_cords_xcycwh) { _bb_cords_xcycwh = std::move(bb_cords_xcycwh); }
+    void set_bb_cords(BoundingBoxCords bb_cords) { _bb_cords = std::move(bb_cords); }
+protected:
+    BoundingBoxCords _bb_cords = {}; // For bb use
+    BoundingBoxCords_xcycwh _bb_cords_xcycwh = {}; // For bb use
+};
+
+struct InstanceSegmentation : public BoundingBox {
```

Comment:
Rename it to PolygonMasks. Instance segmentation is a task, polygon_masks is the metadata structure used in that task

Reply:
Let's rename to PolygonMasks

Reply:
Done